File: TP2-CPP/Kode Program TP 2/Elektronik.cpp
#include <iostream>
#include <string>
#include "Produk.cpp" // Asumsikan file Produk.h sudah dibuat

class Elektronik : public Produk {
protected:
    // Atribut tambahan untuk kelas Elektronik
    std::string lokasi_penyimpanan;
    std::string nama;
    std::string merk;

public:
    // Konstruktor
    Elektronik(std::string id_produk, std::string asal_produk, double harga, std::string lokasi_penyimpanan, std::string nama, std::string merk)
        : Produk(id_produk, asal_produk, harga) { // Memanggil konstruktor kelas induk
        this->lokasi_penyimpanan = lokasi_penyimpanan;
        this->nama = nama;
        this->merk = merk;
    }

    // Getter methods (const)
    std::string getLokasi_penyimpanan() const {
        return lokasi_penyimpanan;
    }

    std::string getNama() const {
        return nama;
    }

    std::string getMerk() const {
        return merk;
    }

};

File: TP2-PHP/Kode Program/Elektronik.php
<?php
require_once 'Produk.php';

class Elektronik extends Produk {
    protected $lokasi_penyimpanan;
    protected $nama;
    protected $merk;

    public function __construct($id_produk, $asal_produk, $harga, $lokasi_penyimpanan, $nama, $merk) {
        parent::__construct($id_produk, $asal_produk, $harga);
        $this->lokasi_penyimpanan = $lokasi_penyimpanan;
        $this->nama = $nama;
        $this->merk = $merk;
    }

    //getter methods
    public function getLokasiPenyimpanan() {
        return $this->lokasi_penyimpanan;
    }

    public function getNama() {
        return $this->nama;
    }

    public function getMerk() {
        return $this->merk;
    }


}
